docs/workspaces.md

```diff
@@ -0,0 +1,87 @@
+# Workspaces
+
+Multi-tenancy is implemented through workspaces. A workspace is a Kubernetes-cluster-like
+HTTPS endpoint, i.e. an endpoint usual Kubernetes client tooling (client-go, controller-runtime
+and others) and user interfaces (kubectl, helm, web console, ...) can talk to like to a
+Kubernetes cluster.
+
+Workspaces can be backed by a traditional REST store implementation through CRDs
+or native resources, persisted in etcd. But there can be alternative implementations
+for special access patterns, e.g. a virtual workspace apiserver that transforms
+other APIs e.g. by projections (Workspace in kcp is a projection of ClusterWorkspace)
+or by applying visibility filters (e.g. showing all workspaces or all namespaces 
+the current user has access to).
+
+Workspaces are represented to the user via the Workspace kind, e.g.
+
+```yaml
+kind: Workspace
+apiVersion: tenancy.kcp.dev/v1beta1
+spec:
+  type: Universal
+status:
+  url: https://kcp.example.com/clusters/myapp
+```
+
+There is a 3-level hierarchy of workspaces:
+
+- **Enduser Workspaces** are workspaces holding enduser resources, e.g.
+  applications with services, secrets, configmaps, deployments, etc.
+- **Organization Workspaces** are workspaces holding organizational data, 
+  e.g. definitions of enduser workspaces, roles, policies, physical cluster
+  registrations, accounting data, etc.
```

Comment:
This is the first time I'm seeing us talk about putting physical clusters in an org workspace and not a user workspace. Does this mean users can't register physical clusters?

Reply:
have removed pclusters from the enumeration. What I meant is that in an org we have those APIs by default, not saying they cannot be in an "enduser workspace".